banyand/liaison/http/server.go

```diff
@@ -134,23 +146,148 @@ func (p *server) GetPort() *uint32 {
 
 func (p *server) PreRun(_ context.Context) error {
 	p.l = logger.GetLogger(p.Name())
+	p.l.Info().Str("level", p.l.GetLevel().String()).Msg("Logger initialized")
+
+	// Log flag values after parsing
+	p.l.Debug().Bool("tls", p.tls).Str("certFile", p.certFile).Str("keyFile", p.keyFile).Msg("Flag values after parsing")
+
+	// Initialize TLSReloader if TLS is enabled
+	p.l.Debug().Bool("tls", p.tls).Msg("HTTP TLS flag is set")
+	if p.tls {
+		p.l.Debug().Str("certFile", p.certFile).Str("keyFile", p.keyFile).Msg("Initializing TLSReloader for HTTP")
+		var err error
+		p.tlsReloader, err = pkgtls.NewReloader(p.certFile, p.keyFile, p.l)
+		if err != nil {
+			p.l.Error().Err(err).Msg("Failed to initialize TLSReloader for HTTP")
+			return err
+		}
+	} else {
+		p.l.Warn().Msg("HTTP TLS is disabled, skipping TLSReloader initialization")
+	}
+
+	// Initialize gRPC client with cert file
+	if p.grpcCert != "" {
```

Comment:
Could you use the reloader for reloading? The reloader can act as either a server or a client. If only credentials are passed to the reloader, it can generate the client-side TLS configuration.